src/components/keywords.jl

```diff
@@ -73,7 +73,7 @@ function parse_kw(ps::ParseState; is_toplevel = false)
     elseif k === Tokens.PRIMITIVE
         return @default ps parse_primitive(ps)
     elseif k === Tokens.PUBLIC
-        if is_toplevel && VERSION > v"1.11-"
+        if is_toplevel && ps.nt.kind !== Tokens.LPAREN && VERSION > v"1.11-"
```

Comment:
Maybe a `!isemptyws(ps.ws)` is more appropriate here?
```
public (foo)
```
shouldn't parse `public` as the kw either, afaict, but would with your implementation.

Reply:
I'm not familiar with the internals though, but the result seems correct？
```julia
julia> CSTParser.parse("public (foo)")
  1:12  errortoken( CSTParser.UnexpectedWhiteSpace)
  1:12   call
  1:7     public
  8:10    foo

julia> CSTParser.parse("public (foo)")[1][1]
  1:7   public

julia> CSTParser.parse("public (foo)")[1][1].head
:IDENTIFIER
```

Reply:
Oh, I kinda butchered that comment, sorry.
My initial thinking was that the WS check should handle almost all cases (including your initial one). `public (foo)` should parse as a kw and not an identifier imho (and I think the reference parser kinda agrees with that), but it's not super important.

With only `is_toplevel && !isemptyws(ps.ws)` we get the following:
```
julia> CSTParser.parse("public(foo)")
  1:11  call
  1:6    public
  7:9    foo

julia> CSTParser.parse("public (foo)")
  1:12  public
  1:5    brackets
  1:3     foo

julia> CSTParser.parse("public foo")
  1:10  public
  1:3    foo

julia> CSTParser.parse("""public"foo" """)
  1:12  macrocall
  1:6    @public_str
  7:6    NOTHING: nothing
  7:12   STRING: foo
```
which seems correct to me (the second case is arguable though).

Reply:
Just found that `public"..."` is not allowed in top-level... Not sure if this is a upstream bug.

Reply:
Certainly looks like one.